src/wallet/wallet.cpp

```diff
@@ -2873,121 +2931,63 @@ bool CWallet::NewKeyPool()
 {
     {
         LOCK(cs_wallet);
+
+        // Check HD Wallet
+        if (IsHDEnabled()) {
+            // Should never happen.
+            LogPrintf("%s::NewKeyPool called from an already enabled HD wallet\n", __func__);
+            return false;
+        }
+
         CWalletDB walletdb(strWalletFile);
-        for (int64_t nIndex : setKeyPool)
+        for (int64_t nIndex : m_spk_man->set_pre_split_keypool)
             walletdb.ErasePool(nIndex);
-        setKeyPool.clear();
+        m_spk_man->set_pre_split_keypool.clear();
 
         if (IsLocked())
             return false;
 
         int64_t nKeys = std::max(GetArg("-keypool", 1000), (int64_t)0);
```

Comment:
reminder: we discussed the lowering of the number of keypairs initially generated once HD functionality was fully implemented. ie, no need to generate 1000 keypairs anymore...reduce to a more "sane" number (100).

Reply:
Actually `DEFAULT_KEYPOOL_SIZE` should be used here.